sdk/python/kfp/v2/compiler_cli_tests/test_data/pipeline_with_nested_conditions_yaml.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from kfp import components
+from kfp import dsl
+import kfp.v2.compiler as compiler
+
+
+def random_num_op(low, high):
+  """Generate a random number between low and high."""
+  return components.load_component_from_text("""
+      name: Generate random number
+      outputs:
+      - {name: output, type: Integer}
+      implementation:
+        container:
+          image: python:alpine3.6
+          command:
+          - sh
+          - -c
+          args:
+          - mkdir -p "$(dirname $2)" && python -c "import random; print(random.randint($0, $1), end='')" | tee $2
+          - "%s"
+          - "%s"
+          - {outputPath: output}
+      """ % (low, high))
```

Comment:
Let's clean this hack later, OK? Injecting constant values in the component text seems pretty unnecessary. Better have them as parameters.

Reply:
Sure, I'll update it later in a separate PR. (As a side note, this was the sample that Hongye reported bugs with)